### PR TITLE
Temp sensor

### PR DESCRIPTION
**Pull request by benjaminwp18**
# Pull Request

## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
Add temperature sensor reading and offset.

### CODE REVIEW

File: src/pi/temp_sensor/README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+## Overview
+
+This package reads from the temperature sensor.
+
+## Installation
+
+```bash
+sudo apt-get install python3-smbus
+```
```
Comment (InvincibleRMC) [inline src/pi/temp_sensor/README.md:11]:
```suggestion

```

File: src/pi/temp_sensor/package.xml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+<?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
+<package format="3">
+  <name>temp_sensor</name>
+  <version>0.0.3</version>
+  <description>Measures temperature</description>
+  <maintainer email="bwp18@case.edu">Benjamin Poulin</maintainer>
+  <license>Apache License 2.0</license>
+
+  <exec_depend>ros2launch</exec_depend>
+
```
Comment (InvincibleRMC) [inline src/pi/temp_sensor/package.xml:11]:
```suggestion

  <exec_depend>ros2launch</exec_depend>
   <exec_depend>python3-smbus</exec_depend>

```

File: src/pi/temp_sensor/temp_sensor/temp_sensor.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import tsys01
+
+READING_TIMER_PERIOD = 0.5  # Seconds
+
+
+class TempSensor(Node):
+
+    def __init__(self) -> None:
+        super().__init__('temp_sensor', parameter_overrides=[])
+        self.publisher = self.create_publisher(Temperature, 'temperature', 10)
```
Comment (InvincibleRMC) [inline src/pi/temp_sensor/temp_sensor/temp_sensor.py:14]: use qos

File: src/surface/gui/gui/widgets/temperature.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+MIN_TEMP_C = 10
+MAX_TEMP_C = 30
+QUEUE_LEN = 5
+
+
+class TemperatureSensor(QWidget):
+    temperature_reading_signal: pyqtSignal = pyqtSignal(Temperature)
+
+    def __init__(self) -> None:
+        super().__init__()
```
Comment (InvincibleRMC) [inline src/surface/gui/gui/widgets/temperature.py:18]:
```suggestion
```

**Review by InvincibleRMC — CHANGES_REQUESTED** [review]

File: src/surface/gui/gui/widgets/temperature.py (hunk 4 above)
Comment (benjaminwp18) [reply src/surface/gui/gui/widgets/temperature.py:18]: ????????????

File: src/surface/gui/gui/widgets/temperature.py [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+        self.temperature_reading_signal.connect(self.temperature_received)
+        self.temp_subscriber: GUIEventSubscriber = GUIEventSubscriber(
+            Temperature,
+            "/tether/temperature",
```
Comment (InvincibleRMC) [inline src/surface/gui/gui/widgets/temperature.py:23]:
```suggestion
            "tether/temperature",
```
And add remapping

File: src/pi/temp_sensor/temp_sensor/temp_sensor.py [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+    def timer_callback(self) -> None:
+        try:
+            self.sensor.read()
+            temp_reading = self.sensor.temperature()
+
+            # If any of the sensors detect water, send true to /tether/flooding
+            msg = Temperature()
+            msg.reading = temp_reading
+            self.publisher.publish(msg)
```
Comment (InvincibleRMC) [inline src/pi/temp_sensor/temp_sensor/temp_sensor.py:30]:
```suggestion
            self.publisher.publish(Temperature(reading=temp_reading))
```

**Review by InvincibleRMC — APPROVED** [review]

**Pull request merged**